clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -520,14 +520,29 @@ void CGDebugInfo::CreateCompileUnit() {
   // the file name itself with no path information. This file name may have had
   // a relative path, so we look into the actual file entry for the main
   // file to determine the real absolute path for the file.
+  // An exception here is workflow when integration footer is involved: in that
+  // case driver passes an absolute path to the original user-provided source
+  // file, whilst main file corresponds to a temporary file generated by the
+  // compiler.
   std::string MainFileDir;
   if (const FileEntry *MainFile = SM.getFileEntryForID(SM.getMainFileID())) {
     MainFileDir = std::string(MainFile->getDir()->getName());
+    FileID MainFileID = SM.getMainFileID();
     if (!llvm::sys::path::is_absolute(MainFileName)) {
       llvm::SmallString<1024> MainFileDirSS(MainFileDir);
       llvm::sys::path::append(MainFileDirSS, MainFileName);
       MainFileName =
           std::string(llvm::sys::path::remove_leading_dotslash(MainFileDirSS));
+    } else if (CGM.getCodeGenOpts().SYCLOverrideMainFileId) {
+      // When integration footer is involved, main file is a temporary file
+      // generated by the compiler, but -main-file-name is expected to contain
+      // an absolute path to the original user-provided source file. We use it
+      // here to properly calculate its checksum.
+      auto ExpectedFileRef = SM.getFileManager().getFileRef(MainFileName);
+      if (ExpectedFileRef) {
```

Comment:
Don't need braces here.

Reply:
Removed in 0a351d1a1b556e14c1dd679bae13ea3e374631dc